bots/ec2_detach_instance_role.py

```diff
@@ -0,0 +1,38 @@
+'''
+## ec2_detach_instance_role
+What it does: Detach an instance role from an EC2 instance. This role needs be passed in through the params.
+Usage: AUTO: ec2_detach_instance_role
+
+Sample GSL: Instance should have roles
+'''
+
+import boto3
+from botocore.exceptions import ClientError
+
+
+def run_action(boto_session,rule,entity,params):
+    text_output = ''
+
+    role_name = entity['name']
+
+    iam = boto_session.client('iam')
+    try:
+        response = iam.list_instance_profiles_for_role(RoleName=role_name)['InstanceProfiles']
+        if len(response) == 0:
+            text_output = 'Role is not attached to instance.\nExiting\n'
+            return text_output
+
+        result = iam.remove_role_from_instance_profile(
+            InstanceProfileName=response[0]['InstanceProfileName'],
+            RoleName=role_name
+        )
+        responseCode = result['ResponseMetadata']['HTTPStatusCode']
+        if responseCode >= 400:

```

Comment:
This check is redundant

Reply:
I check this to avoid errors when I try to get instanceProfile from response. If list is empty I can not use response[0] and the error will be "list index out of range"